pkg/idp/provider/azure/request.go

```diff
@@ -27,10 +28,83 @@ func ProviderScope() []string {
 	return []string{"https://graph.microsoft.com/.default"}
 }
 
-func getPermissionCreateRequestBody(parentApp models.Applicationable) models.Applicationable {
-	app := models.NewApplication()
-	app.SetRequiredResourceAccess(parentApp.GetRequiredResourceAccess())
-	return app
+func computePermissionsUpdatePatch(app models.Applicationable, parentApp models.Applicationable) (bool, []models.RequiredResourceAccessable) {
```

Comment:
I think it could be beneficial to explain what the `parentApp` is used for (maybe in the code as comments or in some readme?). After your explanation I absolutely understand why it is necessary, but I don't think it is self-explanatory, right?